dev-build/cmake/cmake-3.30.3.recipe

```diff
@@ -0,0 +1,356 @@
+
+PROVIDES="
+	cmake$secondaryArchSuffix = $portVersionCompat
+	cmd:ccmake = $portVersionCompat
+	cmd:cmake = $portVersionCompat
+	cmd:cpack = $portVersionCompat
+	cmd:ctest = $portVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libarchive$secondaryArchSuffix
+#	lib:libbz2$secondaryArchSuffix	# not actually used?
+	lib:libcppdap$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libjsoncpp$secondaryArchSuffix
+	lib:libform$secondaryArchSuffix	# from ncurses.
+#	lib:liblzma$secondaryArchSuffix	# not actually used?
+	lib:libncurses$secondaryArchSuffix
+	lib:librhash$secondaryArchSuffix
+	lib:libstdc++$secondaryArchSuffix
+	lib:libuv$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+#	lib:libzstd$secondaryArchSuffix	# not actually used?
+	"
+
+SUMMARY_gui="$SUMMARY (gui)"
+DESCRIPTION_gui="$DESCRIPTION (gui)"
+PROVIDES_gui="
+	cmake${secondaryArchSuffix}_gui = $portVersion
+	cmd:cmake_gui = $portVersionCompat
+	"
+REQUIRES_gui="
+	haiku$secondaryArchSuffix
+	cmake$secondaryArchSuffix == $portVersion base
+	lib:libarchive$secondaryArchSuffix
+	lib:libcppdap$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libjsoncpp$secondaryArchSuffix
+	lib:libQt6Core$secondaryArchSuffix
+	lib:libQt6Gui$secondaryArchSuffix
+	lib:libQt6Widgets$secondaryArchSuffix
+	lib:librhash$secondaryArchSuffix
+	lib:libstdc++$secondaryArchSuffix
+	lib:libuv$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libarchive$secondaryArchSuffix
+#	devel:libbz2$secondaryArchSuffix
+	devel:libcppdap$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libjsoncpp$secondaryArchSuffix >= 25
+#	devel:liblzma$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libQt6Core$secondaryArchSuffix
+	devel:libQt6Gui$secondaryArchSuffix
+	devel:libQt6Widgets$secondaryArchSuffix
+	devel:librhash$secondaryArchSuffix
+	devel:libuv$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+#	devel:libzstd$secondaryArchSuffix
+	devel:nlohmann_json
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+defineDebugInfoPackage cmake$secondaryArchSuffix \
+	$prefix/bin/ccmake \
+	$prefix/bin/cmake \
+	$prefix/bin/cpack \
+	$prefix/bin/ctest \
+	"$(getPackagePrefix gui)/bin"/cmake-gui
+
+BUILD()
+{
+	export LDFLAGS="-lbsd"
+
+	# not an autotools configure
+	./configure --prefix=$prefix \
+		--datadir=/$relativeDataDir/cmake \
+		--docdir=/$relativeDocDir \
+		--mandir=/$relativeManDir \
+		--xdgdatadir=/$relativeDataDir \
+		--system-libs \
+		--qt-gui \
+		--parallel=${jobArgs#-j}
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	# No way to tell this to configure...
+	rm -rf $dataDir/{applications,emacs,icons,mime,vim}
```

Comment:
Will post a small patch to disable installing applications,icons and mime, maybe we could keep the files for emacs and vim?

Reply:
If you really want to disable installing the emacs and vim files you can do so by using:

```diff
diff --git a/Auxiliary/CMakeLists.txt b/Auxiliary/CMakeLists.txt
index c0aebef..43b23c5 100644
--- a/Auxiliary/CMakeLists.txt
+++ b/Auxiliary/CMakeLists.txt
@@ -1,5 +1,6 @@
 # Install Vim files to a typical system integration directory.
 # Packagers can set CMake_INSTALL_VIMFILES_DIR to control this.
+if(NOT HAIKU)
 if(NOT CMake_INSTALL_VIMFILES_DIR)
   set(CMake_INSTALL_VIMFILES_DIR ${CMAKE_XDGDATA_DIR}/vim/vimfiles)
 endif()
@@ -11,6 +12,7 @@ if(NOT CMake_INSTALL_EMACS_DIR)
   set(CMake_INSTALL_EMACS_DIR ${CMAKE_XDGDATA_DIR}/emacs/site-lisp)
 endif()
 install(FILES cmake-mode.el DESTINATION ${CMake_INSTALL_EMACS_DIR})
+endif(NOT HAIKU)
 
 install(FILES cmake.m4 DESTINATION ${CMAKE_XDGDATA_DIR}/aclocal)
 add_subdirectory (bash-completion)
```

Reply:
CMake syntax highlighting in vim is useful.